### DISCUSSION TITLE
Use *.F90 instead of *.f90 suffixes in generated solver code?

### DISCUSSION THREAD

**Discussion by yantosca** [KineticPreProcessor/KPP#26] [Ideas]
In the `GC_updates` branch I had modified KPP to save out the generated solver code with the `.F90` suffixes instead of `.f90`.  The reason being is that there is a Fortran convention that files ending with `.F` or `.F90` contain the raw source code, and files ending with `.f` or `.f90` contain preprocessed code with all of the `#ifdef` and `#include` statements inlined.  

Wanted to get your opinion on whether we should add this modification to our development branch or if we should keep it the way it is.  If we keep it the way it is then we can simply have a build script in the GEOS-Chem repo to move the files from `.f90` to `.F90` (which is what we were doing with older versions of KPP).

Tagging @RolfSander @jimmielin @msl3v @laestrada

**Comment by yantosca** [comment]
Or if there is a way to toggle it so that the user can pick when the mechanism is generated...

**Reply by RolfSander** [reply]
For MECCA, I don't need the feature to create *.F90 instead of *.f90.
Anyway, if this is useful for GEOS-Chem, go ahead and implement it if
you want! (as long as the default remains what we have now)

> Or if there is a way to toggle it so that the user can pick when the
> mechanism is generated...

You could create a new KPP command, maybe call it #UPPERCASEF90. This
has to be implemented in scan.l and scan.y. Then the user could say
either `#UPPERCASEF90 ON` or `#UPPERCASEF90 OFF`. Still, a build script
in the GEOS-Chem repo might be a simpler solution.

**Reply by yantosca** [reply]
Thanks @RolfSander.  I am in the process of bringing in the GEOS-Chem updates to a branch off of dev for evaluation & testing, and will leave `.f90` as the default.  Are there any directions in the PDF manual for adding a new command that would go into the .kpp file?

**Reply by jimmielin** [reply]
> Are there any directions in the PDF manual for adding a new command that would go into the .kpp file?

Hi Bob, you may find this useful: https://github.com/KineticPreProcessor/KPP/commit/43d4a84ecc06c70a41524de8ff0eec45c62b8455

Mike and I added a `#AUTOREDUCE` switch to the `.kpp` file with some edits in `scan.l` and `scan.y`. It should be pretty straightforward to copy the logic to add something similar.

**Reply by RolfSander** [reply]
No, this is not (yet) described in the User Manual. Maybe we should add
a new section called "Instructions for KPP developers"?

Looking at #AUTOREDUCE, as suggested by @jimmielin is a good way to start. It
also shows the necessary changes in scanner.c which I forgot to mention.

I prefer Mike's suggestion of #FPP (or similar) over my clumsy name
#UPPERCASEF90.

**Reply by yantosca** [reply]
Thanks @jimmielin for the suggestion, I'll take a look at what you've done.

**Comment by msl3v** [comment]
F90 is for preprocessing right?
Can you call the command FPP or something?
